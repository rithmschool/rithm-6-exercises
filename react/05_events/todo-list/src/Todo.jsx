import React, { Component } from "react";
import "./Todo.css";

class Todo extends Component {
  constructor(props) {
    super(props);
    this.state = { completed: false };
    this.markComplete = this.markComplete.bind(this);
    this.update = this.update.bind(this);
  }

  markComplete() {
    this.setState({ completed: !this.state.completed });
  }
  isEditing() {}
  update(e) {
    let newText = e.target.parentNodevalue;
    let statei = e.target.getAttribute("index");
    console.log(e.target.parentNode);
    console.log(newText, statei);
    // this.props.handleUpdate(newText, statei);
  }
  render() {
    var status = this.state.completed ? "line-through" : "none";
    return (
      <div className="Todo">
        <input onClick={this.markComplete} type="checkbox" />
        <p style={{ textDecoration: status }}>{this.props.text}</p>
        <input type="text" className="" />
        <button onClick={this.update}>Update</button>
        <button onClick={this.props.handleDelete}>Delete</button>
      </div>
    );
  }
}

export default Todo;
