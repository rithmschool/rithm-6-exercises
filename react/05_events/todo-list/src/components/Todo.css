.Todo {
  display: flex;
  justify-content: space-around;
  align-items: center;
  width: 30rem;
}

.hidden {
  display: hidden;
}

p {
  flex-grow: 2;
}
