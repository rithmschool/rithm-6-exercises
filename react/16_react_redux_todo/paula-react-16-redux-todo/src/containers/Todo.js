// libraries
import React, { Component } from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import PropTypes from "prop-types";

// src
import { DELETE_TODO, TOGGLE_COMPLETE } from "../actions";

const propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  isComplete: PropTypes.bool.isRequired,
  id: PropTypes.string.isRequired,
  dispatch: PropTypes.func.isRequired
};

class Todo extends Component {
  deleteTodo(id) {
    return this.props.dispatch({ type: DELETE_TODO, id });
  }

  toggleComplete(id) {
    return this.props.dispatch({ type: TOGGLE_COMPLETE, id });
  }
  render() {
    const { title, description, isComplete, id } = this.props;

    const strikeStyle = isComplete ? "line-through" : "none";

    return (
      <div>
        <h4 style={{ textDecoration: strikeStyle }}>
          <Link to={`todos/${id}`}>
            {title}: {description}
          </Link>
        </h4>
        <button
          onClick={() => {
            this.toggleComplete(id);
          }}
        >
          Mark as {isComplete ? "incomplete." : "complete!"}
        </button>
        <button
          onClick={() => {
            this.deleteTodo(id);
          }}
        >
          Delete Me!
        </button>
        <button>
          <Link to={`/todos/${id}/edit`}>Edit</Link>
        </button>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    todos: state.todos
  };
}

Todo.propTypes = propTypes;

export default connect(mapStateToProps)(Todo);
