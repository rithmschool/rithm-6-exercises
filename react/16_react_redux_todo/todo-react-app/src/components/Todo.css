* {
  box-sizing: border-box;
}
.completed__look {
  text-decoration-line: line-through;
}

li {
  margin-top: 1%;
}

.button__shape {
  height: 30px;
  width: auto;
  border-radius: 5px;
  margin-left: 5px;
  padding: 5px;
  color: white;
  text-transform: uppercase;
  font-weight: bold;
  font-size: 13px;
}

.delete__todo {
  background-color: darkred;
}

.completed__todo {
  background-color: seagreen;
}

.inprogress__todo {
  background-color: slategray;
}

.edit__todo {
  background-color: goldenrod;
}

.inprogress__edit {
  background-color: dodgerblue;
}
