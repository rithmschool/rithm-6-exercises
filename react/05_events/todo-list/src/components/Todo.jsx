import React, { Component } from "react";
import "./Todo.css";

class Todo extends Component {
  constructor(props) {
    super(props);
    this.state = {
      completed: false,
      isEditing: false
    };
    this.markComplete = this.markComplete.bind(this);
    this.isEditing = this.isEditing.bind(this);
  }

  markComplete() {
    this.setState({ completed: !this.state.completed });
  }
  isEditing(e) {
    e.preventDefault();
    if (this.state.isEditing) {
      let newText = e.target.parentNode.children[2].children[0].value;
      if (newText !== "") {
        let statei = this.props.index;
        this.props.handleUpdate(newText, statei);
      }
    }
    this.setState({ isEditing: !this.state.isEditing });
  }
  render() {
    var status = this.state.completed ? "line-through" : "none";
    var edit = this.state.isEditing ? <input type="text" /> : <br />;
    return (
      <div className="Todo">
        <input onClick={this.markComplete} type="checkbox" />
        <p style={{ textDecoration: status }}>{this.props.text}</p>
        <form action="" onSubmit={this.isEditing}>
          {edit}
        </form>
        <button onClick={this.isEditing}>Update</button>
        <button onClick={this.props.handleDelete}>Delete</button>
      </div>
    );
  }
}

export default Todo;
