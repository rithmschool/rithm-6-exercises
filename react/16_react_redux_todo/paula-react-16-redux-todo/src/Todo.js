// libraries
import React from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";

// src
import "./Todo.css";

const propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  isComplete: PropTypes.bool.isRequired,
  idx: PropTypes.number.isRequired,
  toggleComplete: PropTypes.func.isRequired,
  deleteTodo: PropTypes.func.isRequired
};

const Todo = props => {
  const {
    title,
    description,
    isComplete,
    toggleComplete,
    deleteTodo,
    idx
  } = props;

  const strikeStyle = isComplete ? "line-through" : "none";

  return (
    <div>
      <h4 style={{ textDecoration: strikeStyle }}>
        <Link to={`todos/${idx}`}>
          {title}: {description}
        </Link>
      </h4>
      <button
        onClick={() => {
          toggleComplete(idx);
        }}
      >
        Mark as {isComplete ? "incomplete." : "complete!"}
      </button>
      <button
        onClick={() => {
          deleteTodo(idx);
        }}
      >
        Delete Me!
      </button>
      <button>
        <Link to={`/todos/${idx}/edit`}>Edit</Link>
      </button>
    </div>
  );
};

Todo.propTypes = propTypes;

export default Todo;
