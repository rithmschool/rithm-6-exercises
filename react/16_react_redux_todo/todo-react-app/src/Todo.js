import React from "react";
import "./Todo.css";

const Todo = function({ todo }) {
  let completedTask = todo.isCompleted ? "completed__look" : "";
  let completedBtn = todo.isCompleted ? "completed__todo" : "inprogress__todo";
  const classes = `${completedBtn} button__shape`;
  let completeText = !todo.isCompleted ? "incomplete" : "completed";

  return (
    <li className={completedTask}>
      Title: {todo.title}. Description: {todo.description},{" "}
      <button
        //onClick={handleEdit.bind(this, i)}
        className="edit__todo button__shape"
      >
        Edit Todo{" "}
      </button>
      <button //onClick={handleIsCompleted.bind(this, i)}
        className={classes}
      >
        {completeText}
      </button>
      <button
        className="delete__todo button__shape"
        //onClick={handleDelete.bind(this, i)}
      >
        Delete Todo{" "}
      </button>
    </li>
  );
};

export default Todo;
