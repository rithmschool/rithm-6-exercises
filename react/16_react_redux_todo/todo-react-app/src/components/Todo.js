import React from "react";
import { Link } from "react-router-dom";
import "./Todo.css";

export const Todo = ({ todo, toggleComplete, deleteTodo }) => {
  let completedTask = todo.isCompleted ? "completed__look" : "";
  let completedBtn = todo.isCompleted ? "completed__todo" : "inprogress__todo";
  const classes = `${completedBtn} button__shape`;
  let completeText = !todo.isCompleted ? "incomplete" : "completed";

  return (
    <li className={completedTask}>
      Title: {todo.title}. Description: {todo.description}{" "}
      <Link to={`/todos/${todo.id}/edit`}>
        <button
          //onClick={handleEdit.bind(this, i)}
          className="edit__todo button__shape"
        >
          Edit Todo
        </button>
      </Link>
      <button onClick={() => toggleComplete(todo.id)} className={classes}>
        {completeText}
      </button>
      <button
        className="delete__todo button__shape"
        onClick={() => deleteTodo(todo.id)}
      >
        Delete Todo{" "}
      </button>
    </li>
  );
};

//export default Todo;
