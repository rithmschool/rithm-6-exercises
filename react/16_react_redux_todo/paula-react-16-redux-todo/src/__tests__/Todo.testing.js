import React from "react"; // for JSX
import { shallow, mount } from "enzyme"; // how to mount the component
import { MemoryRouter } from "react-router-dom";
import toJson from "enzyme-to-json"; // for snapshotting the component
import Todo from "../containers/Todo"; // import the component itself

describe("<Todo />", () => {
  let wrapper;
  it("renders", () => {
    // smoke test!!!
    wrapper = shallow(<Todo>hello world</Todo>);
  });
});

describe("<Todo />", () => {
  it("renders", () => {
    let wrapper = shallow(<Todo />);
    console.log(wrapper.debug());
  });
});

describe("<Todo />", () => {
  it("renders", () => {
    let wrapper = mount(<Todo />);
    console.log(wrapper.debug());
  });
});

describe("<Todo />", () => {
  let wrapper;
  it("renders", () => {
    // smoke test!!!
    wrapper = shallow(<Todo>hello world</Todo>);
  });

  it("matches snapshot", () => {
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });
});

describe("<Todo />", () => {
  let wrapper;
  it("renders", () => {
    wrapper = mount(
      <MemoryRouter initialEntries={[{ key: "testKey" }]}>
        <Todo />
      </MemoryRouter>
    );
  });

  it("matches snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
