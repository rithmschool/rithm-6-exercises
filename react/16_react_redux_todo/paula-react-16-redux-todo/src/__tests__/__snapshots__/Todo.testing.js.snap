// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Todo /> matches snapshot 1`] = `undefined`;

exports[`<Todo /> matches snapshot 2`] = `undefined`;
